crates/networking/p2p/peer_channels.rs

```diff
@@ -318,4 +320,112 @@ impl PeerChannels {
         }
         Some((storage_keys, storage_values, should_continue))
     }
+
+    /// Requests state trie nodes given the root of the trie where they are contained and their path (be them full or partial)
+    /// Returns the nodes or None if:
+    /// - There are no available peers (the node just started up or was rejected by all other nodes)
+    /// - The response timed out
+    /// - The response was empty or not valid
```

Comment:
Haven't read the implementation for this, but it's not clear what happens if some nodes are found and others have issues (partial success). Does it return `None` or the nodes that are valid? Seems like the comment should make this clear

Reply:
If a node is invalid then the whole response would be invalid, there are no partial successes.